lib/executor/executors_test.go

```diff
@@ -444,3 +449,68 @@ func TestConfigMapParsingAndValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestArchiveRoundTripScenarioOptions(t *testing.T) {
+	t.Parallel()
+	arc1 := &lib.Archive{
+		Type:      "js",
+		K6Version: consts.Version,
+		Options: lib.Options{
+			VUs:        null.IntFrom(12345),
+			SystemTags: &metrics.DefaultSystemTagSet,
+			Scenarios: map[string]lib.ExecutorConfig{
+				"const-vus": ConstantVUsConfig{
+					BaseConfig: BaseConfig{
+						Name:         "const-vus",
+						Type:         "constant-vus",
+						StartTime:    types.NullDurationFrom(10 * time.Second),
+						GracefulStop: types.NullDurationFrom(30 * time.Second),
+						Env: map[string]string{
+							"FOO": "bar",
+						},
+						Exec: null.StringFrom("default"),
+						Tags: map[string]string{
+							"tagkey": "tagvalue",
+						},
+						Options: lib.ScenarioOptions{
+							Browser: map[string]any{
+								"someOption": "someValue",
+							},
+						},
+					},
+					VUs:      null.IntFrom(50),
+					Duration: types.NullDurationFrom(10 * time.Minute),
+				},
+			},
+		},
+		FilenameURL: &url.URL{Scheme: "file", Path: "/path/to/a.js"},
+		Data:        []byte(`// a contents`),
+		PwdURL:      &url.URL{Scheme: "file", Path: "/path/to"},
+		Filesystems: map[string]fsext.Fs{
+			"file": makeMemMapFs(t, map[string][]byte{
+				"/path/to/a.js": []byte(`// a contents`),
+			}),
+		},
+	}
+
+	buf := bytes.NewBuffer(nil)
+	require.NoError(t, arc1.Write(buf))
+
+	arc2, err := lib.ReadArchive(buf)
+	require.NoError(t, err)
```

Comment:
We are doing more of an integration test here, if I'm getting the code right, why not just test the marshal operation as we did before?

I agree that an archive integration test could be helpful but it should assert the entire scenario config and not only one value.

Reply:
:confused: I don't see it as an integration test, but as an archive unit test, similar to the ones we have [here](https://github.com/grafana/k6/blob/698e894a36bebb60a9895d3abe8e94702b929a3b/lib/archive_test.go#L117). The only reason this is in `lib/executor/` instead is to avoid an import cycle (`lib` -> `lib/executor` -> `lib`), because we need to import a `lib.ExecutorConfig` implementation.

> why not just test the marshal operation as we did before?

As we did before, where?

> I agree that an archive integration test could be helpful but it should assert the entire scenario config and not only one value.

Sure, I can add more scenario assertions, if you want. I didn't consider anything else important, since I only wanted to confirm that the new `options` field is (de)serialized properly.

Reply:
@codebien See 59d17daff.

Reply:
I don't see how the JSON API can be the minimal unit, given that this test is written from the perspective of `lib.Archive`. It just does a round-trip read/write to confirm all options are intact. The JSON API is not involved here at all, and the only "components" are `lib.Archive` and `lib.executor`... :confused: But let's not get into the semantics of an integration/unit test... :sweat_smile: It's really not important in this case, anyway.

I added a comment in 2d82729b2. The import cycle was also mentioned before in the commit message of 07762a22d.

`TestConfigMapParsingAndValidation` only does the unmarshalling process, and not the round-trip. I also thought involving `lib.Archive` was important, as these options need to be present in the archive `metadata.json`, so didn't just want to do `json.Marshal()` on them, but to test it with the actual archive.